src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/ExternalSettingsProviderService.cs

```diff
@@ -6,18 +6,18 @@
 using System;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.Linq;
 using System.Runtime.InteropServices;
 using System.Threading;
 using System.Threading.Tasks;
-//using Microsoft.VisualStudio.Text.Editor.Commanding.Commands;
 using Microsoft.VisualStudio.Utilities.UnifiedSettings;
 using NuGet.Configuration;
 using NuGet.VisualStudio;
 
 namespace NuGet.PackageManagement.VisualStudio.Services
 {
     [Guid("6C09BBE2-4537-48B4-87D8-01BF5EB75901")]
-    public sealed class ExternalSettingsProviderService : IExternalSettingsProvider
+    public sealed class ExternalSettingsProviderService : IExternalSettingsProvider, IExternalArrayItemCommandsProvider
```

Comment:
FWIW, the way we imagined it working is that there would be one external provider implementation per registered external region.  If having a single provider for multiple regions is working for you, that's fine, but it could end up being awkward or sub-optimal in some scenarios.  E.g. error state is per-region, so if a single provider impl handles multiple regions, there's no way for the ErrorConditionResolved event to specify which region it applies to.